lib/features/in_app_review/business/in_app_rating_service.dart

```diff
@@ -0,0 +1,59 @@
+import "package:flutter_riverpod/flutter_riverpod.dart";
+import "package:in_app_review/in_app_review.dart";
+import "package:shared_preferences/shared_preferences.dart";
+
+import "../../../config/shared_prefs.dart";
+import "../../../utils/datetime_utils.dart";
+import "../../app_streak/business/get_days_use_case.dart";
+
+class InAppRatingService {
+  const InAppRatingService(this.ref);
+
+  final Ref ref;
+
+  InAppReview get _inAppReview => InAppReview.instance;
+
+  Future<int> _getUsageDays() async {
+    return await ref.read(getUsageDaysUseCaseProvider.future);
+  }
+
+  SharedPreferences get _sharedPrefs =>
+      ref.read(sharedPreferencesSingletonProvider).requireValue;
+
+  Future<void> requestReviewIfNeeded() async {
+    if (await _canRequestReview()) {
+      await _inAppReview.requestReview();
+      await _updatePreferences();
+    }
+  }
+
+  Future<bool> _canRequestReview() async {
+    final lastReviewPrompt = _sharedPrefs.getInt(
```

Comment:
so this is purely proposition. I am not fully sure if that's the right way, but you can consider using our wrapper on `DateTime`, called `Timestamp` which would look more or less like this:
```dart
 final lastReviewPrompt = Timestamp.tryParse(
      _sharedPrefs.getString(
        // it uses string instead of int, but this doesn't really matter, don't it?
        InAppReviewConfig
            .lastReviewPromptDateInMillis, // this name prolly should be adjusted then
      ),
    );
```
but again, not sure if this gains us a lot of advantage somewhere

Reply:
Now when I think of it, we're using shared prefs and Timestamp in multiple places in the app already. So we could make an extension on the `SharedPrefs` called `getTimestamp` and `saveTimestamp` that would be reusable and would take the serialization/deserialization in one common place

Reply:
Another good suggestion. Making extension here made code even more readable imo ;)